elftools/dwarf/dwarfinfo.py

```diff
@@ -253,23 +336,41 @@ def range_lists(self):
 
     #------ PRIVATE ------#
 
-    def _parse_CUs_iter(self):
+    def _parse_CUs_iter(self, offset=0):
         """ Parse CU entries from debug_info. Yield CUs in order of appearance.
+
+            See .iter_CUs(), .get_CU_containing(), and .get_CU_at().
         """
         if self.debug_info_sec is None:
             return
 
-        offset = 0
         while offset < self.debug_info_sec.size:
-            cu = self._parse_CU_at_offset(offset)
+            cu = self._cached_CU_at_offset(offset)
             # Compute the offset of the next CU in the section. The unit_length
             # field of the CU header contains its size not including the length
             # field itself.
             offset = (  offset +
                         cu['unit_length'] +
                         cu.structs.initial_length_field_size())
+
             yield cu
 
+    def _cached_CU_at_offset(self, offset):
+        """ Lookup cu in the cache.  If not present, first parse and insert it.
+
+            See get_CU_at().
+        """
+        # If the insert point is 0 the entry does not match (perhaps the cache
+        # is empty).  Otherwise on a miss parse and insert.
```

Comment:
I'm not sure I understand what the second sentence in this comment means. Please rephrase.

Reply:
it was saying if the first sentence was false, then do the actions.  Hopefully the new is clearer,